packages/core-design-system/scripts/build.js

```diff
@@ -0,0 +1,287 @@
+  // collect all tokensets for all themes and dedupe
+  const tokensets = [...new Set(Object.values(themes).reduce((acc, sets) => [...acc, ...sets], []))]
+
+  // figure out which tokensets are theme-specific
+  // this is determined by checking if a certain tokenset is used for EVERY theme dimension variant
+  // if it is, then it is not theme-specific
+  const themeableSets = tokensets.filter(set => {
+    return !Object.values(themes).every(sets => sets.includes(set))
+  })
+
+  const configs = Object.entries(themes).map(([theme, sets]) => {
+    return {
+      source: sets.map(tokenset => `design-tokens/${tokenset}.json`),
+      preprocessors: ['tokens-studio'],
+      platforms: {
+        css: {
+          ...COMMON_CONFIG,
+          files: [
+            // core tokens (colors, typography, dimensions, components)
+            ...generateCoreFiles({
+              destination: DESIGN_SYSTEM_ROOT,
+              type: 'css',
+              format: STYLE_BUILD_FORMATS.CSS_VARIABLES
+            }),
+            // semantic tokens, e.g. for application developer
+            ...generateThemeFiles({
+              theme,
+              destination: DESIGN_SYSTEM_ROOT,
+              type: 'css',
+              format: STYLE_BUILD_FORMATS.CSS_VARIABLES
+            })
+          ]
+        },
+        js: {
+          ...COMMON_CONFIG,
+          files: [
+            // core tokens (colors, typography, dimensions, components)
+            ...generateCoreFiles({
+              destination: DESIGN_SYSTEM_ROOT_ESM,
+              type: 'ts',
+              format: STYLE_BUILD_FORMATS.JS_ESM
+            }),
+            // semantic tokens, e.g. for application developer
+            ...generateThemeFiles({
+              theme,
+              destination: DESIGN_SYSTEM_ROOT_ESM,
+              type: 'ts',
+              format: STYLE_BUILD_FORMATS.JS_ESM
+            })
+          ]
+        }
+      }
+    }
+  })
+
+  for (const cfg of configs) {
+    const sd = new StyleDictionary(cfg, {
+      // verbosity: 'verbose'
+    })
+
+    /**
+     * This transform checks for each token whether that token's value could change
+     * due to Tokens Studio theming.
+     * Any tokenset from Tokens Studio marked as "enabled" in the $themes.json is considered
+     * a set in which any token could change if the theme changes.
+     * Any token that is inside such a set or is a reference with a token in that reference chain
+     * that is inside such a set, is considered "themeable",
+     * which means it could change by theme switching.
+     *
+     * This metadata is applied to the token so we can use it as a way of filtering outputs
+     * later in the "format" stage.
+     */
+    sd.registerTransform({
+      name: 'attribute/themeable',
+      type: 'attribute',
+      transform: token => {
+        function isPartOfEnabledSet(token) {
+          const set = token.filePath.replace(/^design-tokens\//g, '').replace(/.json$/g, '')
```

Comment:
nit: Could be simplified to `const set = token.filePath.match(/design-tokens\/(.*)\.json/)?.[1];`